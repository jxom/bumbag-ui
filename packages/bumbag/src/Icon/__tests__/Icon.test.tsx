import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { faArrowLeft, faIgloo } from '@fortawesome/free-solid-svg-icons';
import { Icon } from '../Icon';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Icon icon="info-circle">Hello world</Icon>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(
      <Icon color="primary" icon="info-circle">
        Hello world
      </Icon>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with a label', () => {
    const { container } = render(
      <Icon label="Information" icon="info-circle">
        Hello world
      </Icon>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with aria-hidden', () => {
    const { container } = render(
      <Icon aria-hidden icon="info-circle">
        Hello world
      </Icon>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with a size', () => {
    const { container } = render(
      <Icon fontSize="400" icon="info-circle">
        Hello world
      </Icon>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for direct import', () => {
    const { container } = render(
      <Icon icon={faIgloo} type="font-awesome">
        Hello world
      </Icon>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for custom icon', () => {
    const calendar = {
      viewBoxWidth: 16,
      viewBoxHeight: 16,
      paths: [
        'M11 3c.6 0 1-.5 1-1V1c0-.6-.4-1-1-1s-1 .4-1 1v1c0 .5.4 1 1 1zm3-2h-1v1c0 1.1-.9 2-2 2s-2-.9-2-2V1H6v1c0 1.1-.9 2-2 2s-2-.9-2-2V1H1c-.6 0-1 .5-1 1v12c0 .6.4 1 1 1h13c.6 0 1-.4 1-1V2c0-.6-.5-1-1-1zM5 13H2v-3h3v3zm0-4H2V6h3v3zm4 4H6v-3h3v3zm0-4H6V6h3v3zm4 4h-3v-3h3v3zm0-4h-3V6h3v3zM4 3c.6 0 1-.5 1-1V1c0-.6-.4-1-1-1S3 .4 3 1v1c0 .5.4 1 1 1z',
      ],
    };

    const { container } = render(<Icon icon={calendar}>Hello world</Icon>);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('as', () => {
    it('should render correctly', () => {
      const { container } = render(
        <Icon use="div" icon="info-circle">
          Hello world
        </Icon>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with Icon props', () => {
      const { result } = renderHook(() => Icon.useProps({ icon: 'info-circle' }));
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(
        <Icon icon="info-circle">{(iconProps) => <div {...iconProps}>Hello world</div>}</Icon>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('overrides', () => {
  it('Icon.base should render correctly', () => {
    const { container } = render(
      <Icon icon="info-circle" overrides={{ Icon: { styles: { base: { backgroundColor: 'red' } } } }}>
        hello world
      </Icon>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Icon.iconSets should render correctly', () => {
    const { container } = render(
      <Icon
        icon="solid-arrow-left"
        overrides={{
          Icon: {
            iconSets: [
              {
                icons: [faArrowLeft],
                prefix: 'solid-',
                type: 'font-awesome',
              },
            ],
          },
        }}
      >
        hello world
      </Icon>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Icon.icons should render correctly', () => {
    const { container } = render(
      <Icon
        icon="calendar"
        overrides={{
          Icon: {
            icons: {
              calendar: {
                viewBoxWidth: 16,
                viewBoxHeight: 16,
                paths: [
                  'M11 3c.6 0 1-.5 1-1V1c0-.6-.4-1-1-1s-1 .4-1 1v1c0 .5.4 1 1 1zm3-2h-1v1c0 1.1-.9 2-2 2s-2-.9-2-2V1H6v1c0 1.1-.9 2-2 2s-2-.9-2-2V1H1c-.6 0-1 .5-1 1v12c0 .6.4 1 1 1h13c.6 0 1-.4 1-1V2c0-.6-.5-1-1-1zM5 13H2v-3h3v3zm0-4H2V6h3v3zm4 4H6v-3h3v3zm0-4H6V6h3v3zm4 4h-3v-3h3v3zm0-4h-3V6h3v3zM4 3c.6 0 1-.5 1-1V1c0-.6-.4-1-1-1S3 .4 3 1v1c0 .5.4 1 1 1z',
                ],
              },
            },
          },
        }}
      >
        hello world
      </Icon>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Icon.iconNames should render correctly', () => {
    const { container } = render(
      <Icon
        icon="foo"
        overrides={{
          Icon: {
            iconSets: [
              {
                icons: [faArrowLeft],
                prefix: 'solid-',
                type: 'font-awesome',
              },
            ],
            iconNames: {
              foo: 'solid-arrow-left',
            },
          },
        }}
      >
        hello world
      </Icon>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('Icon.base should render correctly', () => {
    const { container } = render(<Icon icon="info-circle">hello world</Icon>, {
      theme: { Icon: { styles: { base: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Icon.iconSets should render correctly', () => {
    const { container } = render(<Icon icon="solid-arrow-left">hello world</Icon>, {
      theme: {
        Icon: {
          iconSets: [
            {
              icons: [faArrowLeft],
              prefix: 'solid-',
              type: 'font-awesome',
            },
          ],
        },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Icon.icons should render correctly', () => {
    const { container } = render(<Icon icon="calendar">hello world</Icon>, {
      theme: {
        Icon: {
          icons: {
            calendar: {
              viewBoxWidth: 16,
              viewBoxHeight: 16,
              paths: [
                'M11 3c.6 0 1-.5 1-1V1c0-.6-.4-1-1-1s-1 .4-1 1v1c0 .5.4 1 1 1zm3-2h-1v1c0 1.1-.9 2-2 2s-2-.9-2-2V1H6v1c0 1.1-.9 2-2 2s-2-.9-2-2V1H1c-.6 0-1 .5-1 1v12c0 .6.4 1 1 1h13c.6 0 1-.4 1-1V2c0-.6-.5-1-1-1zM5 13H2v-3h3v3zm0-4H2V6h3v3zm4 4H6v-3h3v3zm0-4H6V6h3v3zm4 4h-3v-3h3v3zm0-4h-3V6h3v3zM4 3c.6 0 1-.5 1-1V1c0-.6-.4-1-1-1S3 .4 3 1v1c0 .5.4 1 1 1z',
              ],
            },
          },
        },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Icon.iconNames should render correctly', () => {
    const { container } = render(<Icon icon="foo">hello world</Icon>, {
      theme: {
        Icon: {
          iconSets: [
            {
              icons: [faArrowLeft],
              prefix: 'solid-',
              type: 'font-awesome',
            },
          ],
          iconNames: {
            foo: 'solid-arrow-left',
          },
        },
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<Icon icon="info-circle">hello world</Icon>, {
      theme: { Icon: { defaultProps: { className: 'test' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for color', () => {
    const { container } = render(<Icon icon="info-circle">hello world</Icon>, {
      theme: { Icon: { defaultProps: { color: 'primary' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for label', () => {
    const { container } = render(<Icon icon="info-circle">hello world</Icon>, {
      theme: { Icon: { defaultProps: { label: 'Circle' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for fontSize', () => {
    const { container } = render(<Icon icon="info-circle">hello world</Icon>, {
      theme: { Icon: { defaultProps: { fontSize: '300' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
