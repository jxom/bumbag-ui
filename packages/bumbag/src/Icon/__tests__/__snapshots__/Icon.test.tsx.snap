// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition as should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
  role="img"
  viewbox="0 0 512 512"
>
  <path
    d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
    fill="currentColor"
    fill-rule="evenodd"
  />
</div>
`;

exports[`composition hook should return with Icon props 1`] = `
Object {
  "children": <React.Fragment />,
  "className": "css-i5runh css-1pwrqjj css-1yuzjtp fp-Box fp-Icon",
  "role": "img",
  "viewBox": "0 0 0 0",
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
  role="img"
  viewbox="0 0 512 512"
>
  Hello world
</div>
`;

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

<svg
  class="emotion-0 emotion-1 emotion-2 test fp-Box fp-Icon"
  role="img"
  viewBox="0 0 512 512"
>
  <path
    d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
    fill="currentColor"
    fill-rule="evenodd"
  />
</svg>
`;

exports[`defaultProps should render correctly for color 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2 {
  color: #574feb !important;
}

.emotion-0 {
  fill: #574feb;
  position: relative;
  height: 1em;
  width: 1em;
}

<svg
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
  role="img"
  viewBox="0 0 512 512"
>
  <path
    d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
    fill="currentColor"
    fill-rule="evenodd"
  />
</svg>
`;

exports[`defaultProps should render correctly for fontSize 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

.emotion-2 {
  font-size: 1.25rem !important;
}

<svg
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
  role="img"
  viewBox="0 0 512 512"
>
  <path
    d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
    fill="currentColor"
    fill-rule="evenodd"
  />
</svg>
`;

exports[`defaultProps should render correctly for label 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

<svg
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
  label="Circle"
  role="img"
  viewBox="0 0 512 512"
>
  <title>
    Circle
  </title>
  <path
    d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
    fill="currentColor"
    fill-rule="evenodd"
  />
</svg>
`;

exports[`overrides Icon.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

.emotion-0 {
  background-color: #ff0000;
}

<svg
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
  role="img"
  viewBox="0 0 512 512"
>
  <path
    d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
    fill="currentColor"
    fill-rule="evenodd"
  />
</svg>
`;

exports[`overrides Icon.iconNames should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

<svg
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
  role="img"
  viewBox="0 0 0 0"
/>
`;

exports[`overrides Icon.iconSets should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

<svg
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
  role="img"
  viewBox="0 0 0 0"
/>
`;

exports[`overrides Icon.icons should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

<svg
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
  role="img"
  viewBox="0 0 0 0"
/>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

<svg
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
  role="img"
  viewBox="0 0 512 512"
>
  <path
    d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
    fill="currentColor"
    fill-rule="evenodd"
  />
</svg>
`;

exports[`props should render correctly for custom icon 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

<svg
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
  role="img"
  viewBox="0 0 16 16"
>
  <path
    d="M11 3c.6 0 1-.5 1-1V1c0-.6-.4-1-1-1s-1 .4-1 1v1c0 .5.4 1 1 1zm3-2h-1v1c0 1.1-.9 2-2 2s-2-.9-2-2V1H6v1c0 1.1-.9 2-2 2s-2-.9-2-2V1H1c-.6 0-1 .5-1 1v12c0 .6.4 1 1 1h13c.6 0 1-.4 1-1V2c0-.6-.5-1-1-1zM5 13H2v-3h3v3zm0-4H2V6h3v3zm4 4H6v-3h3v3zm0-4H6V6h3v3zm4 4h-3v-3h3v3zm0-4h-3V6h3v3zM4 3c.6 0 1-.5 1-1V1c0-.6-.4-1-1-1S3 .4 3 1v1c0 .5.4 1 1 1z"
    fill="currentColor"
    fill-rule="evenodd"
  />
</svg>
`;

exports[`props should render correctly for direct import 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

<svg
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
  role="img"
  type="font-awesome"
  viewBox="0 0 576 512"
>
  <path
    d="M320 33.9c-10.5-1.2-21.2-1.9-32-1.9-99.8 0-187.8 50.8-239.4 128H320V33.9zM96 192H30.3C11.1 230.6 0 274 0 320h96V192zM352 39.4V160h175.4C487.2 99.9 424.8 55.9 352 39.4zM480 320h96c0-46-11.1-89.4-30.3-128H480v128zm-64 64v96h128c17.7 0 32-14.3 32-32v-96H411.5c2.6 10.3 4.5 20.9 4.5 32zm32-192H128v128h49.8c22.2-38.1 63-64 110.2-64s88 25.9 110.2 64H448V192zM0 448c0 17.7 14.3 32 32 32h128v-96c0-11.1 1.9-21.7 4.5-32H0v96zm288-160c-53 0-96 43-96 96v96h192v-96c0-53-43-96-96-96z"
    fill="currentColor"
    fill-rule="evenodd"
  />
</svg>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2 {
  color: #574feb !important;
}

.emotion-0 {
  fill: #574feb;
  position: relative;
  height: 1em;
  width: 1em;
}

<svg
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
  role="img"
  viewBox="0 0 512 512"
>
  <path
    d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
    fill="currentColor"
    fill-rule="evenodd"
  />
</svg>
`;

exports[`props should render correctly with a label 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

<svg
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
  label="Information"
  role="img"
  viewBox="0 0 512 512"
>
  <title>
    Information
  </title>
  <path
    d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
    fill="currentColor"
    fill-rule="evenodd"
  />
</svg>
`;

exports[`props should render correctly with a size 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

.emotion-2 {
  font-size: 1.5rem !important;
}

<svg
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
  role="img"
  viewBox="0 0 512 512"
>
  <path
    d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
    fill="currentColor"
    fill-rule="evenodd"
  />
</svg>
`;

exports[`props should render correctly with aria-hidden 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

<svg
  aria-hidden="true"
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
  role="img"
  viewBox="0 0 512 512"
>
  <path
    d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
    fill="currentColor"
    fill-rule="evenodd"
  />
</svg>
`;

exports[`theming Icon.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

.emotion-0 {
  background-color: #ff0000;
}

<svg
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
  role="img"
  viewBox="0 0 512 512"
>
  <path
    d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
    fill="currentColor"
    fill-rule="evenodd"
  />
</svg>
`;

exports[`theming Icon.iconNames should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

<svg
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
  role="img"
  viewBox="0 0 448 512"
>
  <path
    d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
    fill="currentColor"
    fill-rule="evenodd"
  />
</svg>
`;

exports[`theming Icon.iconSets should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

<svg
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
  role="img"
  viewBox="0 0 448 512"
>
  <path
    d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
    fill="currentColor"
    fill-rule="evenodd"
  />
</svg>
`;

exports[`theming Icon.icons should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

<svg
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Icon"
  role="img"
  viewBox="0 0 16 16"
>
  <path
    d="M11 3c.6 0 1-.5 1-1V1c0-.6-.4-1-1-1s-1 .4-1 1v1c0 .5.4 1 1 1zm3-2h-1v1c0 1.1-.9 2-2 2s-2-.9-2-2V1H6v1c0 1.1-.9 2-2 2s-2-.9-2-2V1H1c-.6 0-1 .5-1 1v12c0 .6.4 1 1 1h13c.6 0 1-.4 1-1V2c0-.6-.5-1-1-1zM5 13H2v-3h3v3zm0-4H2V6h3v3zm4 4H6v-3h3v3zm0-4H6V6h3v3zm4 4h-3v-3h3v3zm0-4h-3V6h3v3zM4 3c.6 0 1-.5 1-1V1c0-.6-.4-1-1-1S3 .4 3 1v1c0 .5.4 1 1 1z"
    fill="currentColor"
    fill-rule="evenodd"
  />
</svg>
`;
